docs(terraform_validate): add note about cache and parallelism (#642)
antonbabenko/pre-commit-terraform: README.md

### terraform_validate

> [!IMPORTANT]

> > If you have the [plugin cache](https://developer.hashicorp.com/terraform/cli/config/config-file#provider-plugin-cache) enabled, we recommend disabling parallelism (`--hook-config=--parallelism-limit=1`) to avoid races when `terraform init` writes to it.

Reviewer: ```suggestion
> If you have the [plugin cache](https://developer.hashicorp.com/terraform/cli/config/config-file#provider-plugin-cache) enabled, we recommend enabling `- --hook-config=--retry-once-with-cleanup=true` or disabling parallelism (`--hook-config=--parallelism-limit=1`) to avoid races when `terraform init` writes to it.
```

Reviewer 2: * +1 for `race conditions` (was thinking to add suggestion, though got distracted by something else)
* -1 for `use` instead of `have` (it's `have ... enabled` which is correct)

Reviewer 2: ```suggestion
> If you have the [plugin_cache_dir](https://developer.hashicorp.com/terraform/cli/config/config-file#provider-plugin-cache) enabled, we recommend enabling re-try with cleanup (`--hook-config=--retry-once-with-cleanup=true`) or disabling parallelism (`--hook-config=--parallelism-limit=1`) to avoid race conditions when `terraform init` writes to it.
```

Reviewer: a, okay

Reviewer 2: 🤪

Reviewer: I don't think that `"re-try with cleanup"` makes any sense as  it literally same as `retry-once-with-cleanup=true`

And `plugin_cache_dir` is a CLI option (and part of TF_PLUGIN_CACHE_DIR), so I prefer it than `provider plugin cache`. Or replace it to `TF_PLUGIN_CACHE_DIR`

Reviewer 2: > I don't think that `"re-try with cleanup"` makes any sense as it literally same as `retry-once-with-cleanup=true`

Was just thinking of consistency with the other part of the clause 🤷🏻 Up to you though.

> And `plugin_cache_dir` is a CLI option (and part of TF_PLUGIN_CACHE_DIR), so I prefer it than `provider plugin cache`. Or replace it to `TF_PLUGIN_CACHE_DIR`

Is a CLI option that enables provide plugin cache. Up to you as well though.

> > If you use [`TF_PLUGIN_CACHE_DIR`](https://developer.hashicorp.com/terraform/cli/config/config-file#provider-plugin-cache), we recommend enabling `--hook-config=--retry-once-with-cleanup=true` or disabling parallelism (`--hook-config=--parallelism-limit=1`) to avoid race conditions when `terraform init` writes to it.

Reviewer: ```suggestion
> If you use [`TF_PLUGIN_CACHE_DIR`](https://developer.hashicorp.com/terraform/cli/config/config-file#provider-plugin-cache), we recommend enabling `--hook-config=--retry-once-with-cleanup=true` or disabling parallelism (`--hook-config=--parallelism-limit=1`) to avoid [race conditions when `terraform init` writes to it](https://github.com/hashicorp/terraform/issues/31964).
```

> > If you use the [`TF_PLUGIN_CACHE_DIR`](https://developer.hashicorp.com/terraform/cli/config/config-file#provider-plugin-cache), we recommend enabling `--hook-config=--retry-once-with-cleanup=true` or disabling parallelism (`--hook-config=--parallelism-limit=1`) to avoid race conditions when `terraform init` writes to it.

Reviewer: ```suggestion
> If you use [`TF_PLUGIN_CACHE_DIR`](https://developer.hashicorp.com/terraform/cli/config/config-file#provider-plugin-cache), we recommend enabling `--hook-config=--retry-once-with-cleanup=true` or disabling parallelism (`--hook-config=--parallelism-limit=1`) to avoid race conditions when `terraform init` writes to it.
```

> > If you use the [plugin_cache_dir](https://developer.hashicorp.com/terraform/cli/config/config-file#provider-plugin-cache) enabled, we recommend enabling `--hook-config=--retry-once-with-cleanup=true` or disabling parallelism (`--hook-config=--parallelism-limit=1`) to avoid race conditions when `terraform init` writes to it.

Reviewer: ```suggestion
> If you have the [plugin_cache_dir](https://developer.hashicorp.com/terraform/cli/config/config-file#provider-plugin-cache) enabled, we recommend enabling `--hook-config=--retry-once-with-cleanup=true` or disabling parallelism (`--hook-config=--parallelism-limit=1`) to avoid race conditions when `terraform init` writes to it.
```

> > If you have the [plugin_cache_dir](https://developer.hashicorp.com/terraform/cli/config/config-file#provider-plugin-cache) enabled, we recommend enabling `--hook-config=--retry-once-with-cleanup=true` or disabling parallelism (`--hook-config=--parallelism-limit=1`) to avoid race conditions when `terraform init` writes to it.

Reviewer: ```suggestion
> If you use the [`TF_PLUGIN_CACHE_DIR`](https://developer.hashicorp.com/terraform/cli/config/config-file#provider-plugin-cache), we recommend enabling `--hook-config=--retry-once-with-cleanup=true` or disabling parallelism (`--hook-config=--parallelism-limit=1`) to avoid race conditions when `terraform init` writes to it.
```
